src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbStreamsBySharedTable.java

```diff
@@ -41,22 +88,132 @@ public MtAmazonDynamoDbStreamsBySharedTable(AmazonDynamoDBStreams amazonDynamoDb
      */
     @Override
     public ListStreamsResult listStreams(ListStreamsRequest listStreamsRequest) {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("listStreams request={}", listStreamsRequest);
+        }
+
         checkArgument(mtDynamoDb.getMtContext().getContext() == null,
                 "listStreams for shared table currently does not support calling with tenant context");
         checkArgument(listStreamsRequest.getTableName() == null,
                 "listStreams for shared table currently does not support filtering by table name");
 
+        // filter to mt tables
         ListStreamsResult result = super.listStreams(listStreamsRequest);
```

Comment:
So to summarize, this code gets all streams, then gets the physical tables associated with this SharedTables instance, and filters down to the list of streams for those shared tables only?

Reply:
Yes, exactly.

We could enable this method to work in the context of a tenant in the future.